packages/editor/src/client/blocks/mapLeafletBlock.tsx

```diff
@@ -0,0 +1,168 @@
+import React, {useState} from 'react'
+import {Map, TileLayer, Marker, Popup} from 'react-leaflet'
+import 'leaflet/dist/leaflet.css'
+import '../customCSS/leaflet.css'
+
+import {Icon} from 'leaflet'
+
+// stupid hack so that leaflet's images work after going through webpack
+//@ts-ignore
+import marker from 'leaflet/dist/images/marker-icon.png'
+//@ts-ignore
+import marker2x from 'leaflet/dist/images/marker-icon-2x.png'
+//@ts-ignore
+import markerShadow from 'leaflet/dist/images/marker-shadow.png'
+
+//@ts-ignore
+delete Icon.Default.prototype._getIconUrl
+
+Icon.Default.mergeOptions({
+  iconRetinaUrl: marker2x,
+  iconUrl: marker,
+  shadowUrl: markerShadow
+})
+
+import {
+  BlockProps,
+  FieldProps,
+  Box,
+  Card,
+  PlaceholderInput,
+  ZIndex,
+  IconButton,
+  Spacing,
+  TypographicTextArea,
+  Drawer,
+  Image
+} from '@karma.run/ui'
+
+import {
+  MaterialIconImageOutlined,
+  MaterialIconEditOutlined,
+  MaterialIconClose
+} from '@karma.run/icons'
+
+import {ImageSelectPanel} from '../panel/imageSelectPanel'
+import {ImagedEditPanel} from '../panel/imageEditPanel'
+
+import {MapLeafletBlockValue, MapLeafletItem} from './types'
+
+// import {isFunctionalUpdate} from '@karma.run/react'
+
+export interface MapLeafletBlockProps extends BlockProps<MapLeafletBlockValue> {}
+
+export function MapLeafletBlock({value, onChange, disabled}: BlockProps<MapLeafletBlockValue>) {
+  const {zoom, centerLat, centerLng, caption} = value
+
+  return (
+    <>
+      <Map
+        center={[centerLat, centerLng]}
+        zoom={zoom}
+        style={{width: '100%', height: '45vh'}}
+        onViewportChange={e => {
+          if (e.center && e.zoom) {
+            onChange({...value, centerLat: e.center[0], centerLng: e.center[1], zoom: e.zoom})
+          }
+        }}>
+        <TileLayer
+          url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
+          attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
+        />
+        <Marker position={[centerLat, centerLng]}></Marker>
+      </Map>
+      <Box marginTop={Spacing.ExtraSmall}>
+        <TypographicTextArea
+          variant="subtitle2"
+          align="center"
+          placeholder="Caption"
+          value={caption}
+          onChange={e => {
+            onChange({...value, caption: e.target.value})
+          }}
+        />
+      </Box>
+    </>
+  )
+}
+
+export function MapLeafletItemElement({value, onChange}: FieldProps<MapLeafletItem>) {
```

Comment:
Is this already used?

Reply:
Not yet, it will be needed in the next step when we implement the items. Should I uncomment it meanwhile?

Reply:
In general don't commit code that is not used. If you have a good reason add the reason in a todo.